core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -202,7 +202,7 @@ private[spark] class PythonRDD(
       this.interrupt()
     }
 
-    override def run(): Unit = Utils.logUncaughtExceptions {
+    override def run(): Unit = Utils.tryLog {
```

Comment:
Instead of doing a big tryLog (which is sort of dangerous because of the potential of users adding a return statement that would break this code), can we add an explicit try around the two `worker.shutdownOutput()` calls?